src/alpaka_job_coverage/versions.py

```diff
@@ -0,0 +1,89 @@
+from typing import Dict, List
+
+from alpaka_job_coverage.globals import *  # pylint: disable=wildcard-import,unused-wildcard-import
+from packaging import version as pk_version
+from typeguard import typechecked
+
+# This dict contains all software version, which we handle we the filter rules.
+# Not every software version results in a case in the filter rules. For example
+# we have no case, where a combination with a specific Boost version needs to be
+# disabled (date: 26.7.2023). Maybe in future, a Boost specific case will be
+# added.
+versions: Dict[str, List[str]] = {
+    GCC: ["7", "8", "9", "10", "11", "12"],
```

Comment:
Actual yes. René and I decided to keep all old rules for the moment. If they provide some problems (code complexity, performance issues ...), we will remove it. Therefore, I have to add GCC 6 here because of the existing rule.